server/not-found.js

```diff
@@ -21,10 +21,75 @@ const express = require('express');
 const path = require('path');
 const fs = require('fs');
 
+const {doRedirect} = require('./env');
+const {defaultLocale} = require('../site/_filters/i18n');
+
+const ROOT_DIR = 'dist';
+
+/**
+ * @param {string} originalPath The full URL path, which might include an i18n
+ * prefix.
+ * @param {string} [rootDir] The root for static files. Defaults to 'dist'.
+ * @param {string} [locale] The locale to used. Defaults to 'en'.
+ * @returns {string|null} The full URL path with the i18n prefix removed, if
+ * that would return valid HTML, or null if there is no valid HTML at that
+ * location either.
+ **/
+const getNonLocalizedURL = (
+  originalPath,
+  rootDir = ROOT_DIR,
+  locale = defaultLocale
+) => {
+  // E.g. '/ja/docs/privacy-sandbox/chips/' is split to
+  // ['ja', 'docs', 'privacy-sandbox', 'chips', '']
+  const pathParts = originalPath.substring(1).split('/');
+
+  if (pathParts[0] === locale) {
+    // If the default prefix is already at the start of the path, return early.
+    return null;
+  }
+
+  // If pathParts does not already include a locale prefix, chances are that
+  // assigning 'en' as the first item will result in a path that doesn't exist.
+  // That's okay for our purposes.
+  pathParts[0] = locale;
+
+  // Normalize by removing a possible trailing 'index.html'.
+  if (pathParts[pathParts.length - 1] === 'index.html') {
+    pathParts[pathParts.length - 1] = '';
+  }
+
+  const possibleIndexPath = path.join(rootDir, ...pathParts, 'index.html');
+  try {
+    // This throws if the path is invalid, and returns undefined if it's valid.
+    // (existsSync() is deprecated.)
+    fs.accessSync(possibleIndexPath);
+    // We have a valid file!
+    // Setting the first item to '' will remove the default locale prefix from
+    // the URL we redirect to, while ensuring the URL starts with '/'.
+    pathParts[0] = '';
+    return pathParts.join('/');
+  } catch (err) {
+    // There was no index.html at the default locale path.
+    return null;
+  }
+};
+
 /**
  * @type {express.RequestHandler}
  */
 const notFoundHandler = (req, res, next) => {
+  // Instead of returning a 404, check to see if the request could be fulfilled
+  // by redirecting to the same URL without a locale prefix.
+  // See https://github.com/GoogleChrome/developer.chrome.com/issues/2398
+  const nonLocalizedURL = getNonLocalizedURL(req.path);
+  if (nonLocalizedURL !== null) {
+    // Use a 302 redirect here, since there might be a localized version of this
+    // URL at some point in the future.
+    return doRedirect(res, nonLocalizedURL, 302);
+  }
+
+  // Otherwise, we have a 404, not a redirect.
   res.status(404);
```

Comment:
You need to exempt here a robots.txt file, or, alternatively, provide a valid robots.txt at the root of the project /dist

Reply:
Done, in https://github.com/GoogleChrome/developer.chrome.com/pull/2517